### PR TITLE
Add ListedEquities and Nasdaq

### PR DESCRIPTION
**Pull request by EricForgy**

### CODE REVIEW

**Comment by EricForgy** [comment]
This needs more work before merging, but submitting a PR to start reviewing.

**Comment by felipenoris** [comment]
Why add "database data" into a dep?

**Comment by EricForgy** [comment]
Hey @felipenoris 👋 

It's good to hear from you 😊 

@ScottPJones and I were just talking about you and all the good stuff you've done (and are doing) 👍 

I'm definitely open to suggestions if you have better ideas, but one motivation is to have these tickers available offline.

**Comment by felipenoris** [comment]
@EricForgy , nice to hear that my code is useful to you. That's the whole point indeed.
My point here is that tickers change every day. This kind of data is better kept "outside code".
In finance we're always depending on some sort of database. Of course a package without an embedded database is less useful, but it doesn't look practical to be responsible for keeping this kind of database up to date everyday (as a package owner), so I would just lay out some abstraction in the package and let the user hook it into a database.

**Comment by ScottPJones** [comment]
@felipenoris The codes for stocks, like "MSFT", change daily?

**Comment by EricForgy** [comment]
Individual tickers should not change often, but the list of all stocks will change frequently, e.g. every time there is an IPO, a name is added, and some stocks get delisted as well.

**Comment by felipenoris** [comment]
> @felipenoris The codes for stocks, like "MSFT", change daily?

Hi @ScottPJones! They do change. Might be a rare event. When this things happen, (in my day job) I must solve this kind of issue immediately, **in the morning of the day the change occurs**. And this is why it is difficult to commit this to a package version.

File: src/ListedEquities/Nasdaq.jl [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        response = JSON3.read(read(jsonfile, String))
+    else
+        response = JSON3.read(HTTP.get("http://localhost:$(port)/api/nasdaq/list").body)
+        open(io -> print(io,JSON3.write(response)), jsonfile, "w")
+    end
+    for stock in response
+        s = stock["Symbol"]
+        d = stock["Description"]
+        @eval Nasdaq begin
+            $(Symbol(s)) = ListedEquity{:NASDAQ,Symbol($s)}()
```
Comment (ScottPJones) [inline src/ListedEquities/Nasdaq.jl]: Shouldn't this be `const`?

Comment (EricForgy) [reply src/ListedEquities/Nasdaq.jl]: Yes. Good catch.

**Pull request merged**